### ISSUE TITLE
Baresip problems with sip2sip

### ISSUE THREAD

**Issue by smathieson** [alfredh/baresip#742]
Hello,

I am trying to use a sip2sip account with baresip, configured thusly:
<sip:xxx@sip2sip.info>;answermode=auto;auth_user=xxx@sip2sip.info;auth_pass=xxx;outbound=sip:proxy.sipthor.net;transport=udp

However, while outgoing audio works just fine, incoming audio (from sip2sip) is 'garbled'. I am using a standard config file, nothing changed from a debian install. I am attaching the file.

This ONLY happens when the call is initiated from sip2sip. If baresip initiates the call all works as expected.

Is something wrong or missing? I would have posted to the mailing list but have been unable to subscribe. Sorry.

Thanks,
Sam.
[config.txt](https://github.com/alfredh/baresip/files/3332772/config.txt)

**Comment by juha-h** [comment]
Check that in config g711.so module is loaded and add ;audio_codecs=PCMU,PCMA to the account.

**Comment by smathieson** [comment]
Thanks for the help. I have all the audio codecs loaded. Also tried with just the 711 codec loaded. Wireshark seems to show 711 audio is coming in as expected. Seems like the problem is when the audio is in baresip.

**Comment by alfredh** [comment]
what is the name/version of the SIP client that you are calling from ?

can you please include here the full baresip output (with -v) and also
the Wireshark trace ?

please also enable RTP statistics in your `config` file:

`rtp_stats    yes`

during a call you can use the `/audio_debug` (`A`) command to show
the audio stats.

**Comment by juha-h** [comment]
smathieson writes:

> I am trying to use a sip2sip account with baresip, configured thusly:
> <sip:xxx@sip2sip.info>;answermode=auto;auth_user=xxx@sip2sip.info;auth_pass=xxx;outbound=sip:proxy.sipthor.net;transport=udp
> 
> However, while outgoing audio works just fine, incoming audio (from
> sip2sip) is 'garbled'. I am using a standard config file, nothing
> changed from a debian install. I am attaching the file.
> 
> Is something wrong or missing? I would have posted to the mailing list
> but have been unable to subscribe. Sorry.

I tried with my sip2sip account using both debian baresip and my android
baresip app, and there was no audio issues with pcma nor any aother
codec.  So I don't think this is a general problem with baresip.

**Comment by alfredh** [comment]
I have also tested with sip2sip and it was working fine.

**Comment by smathieson** [comment]
Thanks for the test. I have downloaded and compiled the latest version from this website and I still had the same issue. I did some more testing and then I tried a different client, still no luck... Well long story short... I tried my wife's motorola phone and bingo. No problems with anything.

So the real culprit is.... My Samsung S10. I tried different apps on it and no luck. But it seems this only happens with baresip, since it works from S10 -> linphone. Don't really know what is going on. Either way, it seems as if this issue is not related directly to baresip. Sorry about that.
